device/core/test/_module_client_test.js

```diff
@@ -0,0 +1,155 @@
+        varIndex++;
+      });
+
+      afterEach(function() {
+        // delete all the vars
+        for (var index = 0; index < requiredVars.length; index++) {
+          delete process.env[requiredVars[index]];
+        }
+      });
+
+      requiredVars.forEach(function(_, index) {
+        it('throws if env var ' + requiredVars[index] + ' is not defined', function() {
+          assert.throws(function() {
+            ModuleClient.fromEnvironment(function() {});
+          }, errors.PreconditionFailedError);
+        });
+      });
+    });
+
+    // Tests_SRS_NODE_MODULE_CLIENT_13_030: [ The value for the environment variable IOTEDGE_AUTHSCHEME must be SasToken. ]
+    describe('check IOTEDGE_AUTHSCHEME', function() {
+      var requiredVars = ['IOTEDGE_WORKLOADURI', 'IOTEDGE_DEVICEID', 'IOTEDGE_MODULEID', 'IOTEDGE_IOTHUBHOSTNAME', 'IOTEDGE_MODULEGENERATIONID'];
+
+      beforeEach(function() {
+        for (var index = 0; index < requiredVars.length; index++) {
+          process.env[requiredVars[index]] = '42';
+        }
+      });
+
+      afterEach(function() {
+        // delete all the vars
+        for (var index = 0; index < requiredVars.length; index++) {
+          delete process.env[requiredVars[index]];
+        }
+        delete process.env['IOTEDGE_AUTHSCHEME'];
+      });
+
+      it('throws if value is not SasToken', function() {
+        assert.throws(function() {
+          process.env['IOTEDGE_AUTHSCHEME'] = 'NotSasToken';
+          ModuleClient.fromEnvironment(function() {});
+        }, errors.PreconditionFailedError);
+      });
+    });
+
+    describe('check create', function() {
+      var env = [
+        ['IOTEDGE_WORKLOADURI', 'unix:///var/run/iotedge.w.sock'],
+        ['IOTEDGE_DEVICEID', 'd1'],
+        ['IOTEDGE_MODULEID', 'm1'],
+        ['IOTEDGE_IOTHUBHOSTNAME', 'host1'],
+        ['IOTEDGE_AUTHSCHEME', 'SasToken'],
+        ['IOTEDGE_GATEWAYHOSTNAME', 'gwhost'],
+        ['IOTEDGE_MODULEGENERATIONID', 'g1']
+      ];
+
+      beforeEach(function() {
+        env.forEach(function(e) {
+          process.env[e[0]] = e[1];
+        });
+      });
+
+      afterEach(function() {
+        env.forEach(function(e) {
+          delete process.env[e[0]];
+        });
+      });
+
+      // Tests_SRS_NODE_MODULE_CLIENT_13_032: [ The fromEnvironment method shall create a new IotEdgeAuthenticationProvider object and pass this to the transport constructor. ]
+      // Tests_SRS_NODE_MODULE_CLIENT_13_031: [ The fromEnvironment method shall return a new instance of the ModuleClient object. ]
+      it('creates IotEdgeAuthenticationProvider', function() {
+        var transportStub = sinon.stub().returns(new EventEmitter());
+        var client = ModuleClient.fromEnvironment(transportStub);
+        assert.equal(transportStub.called, true);
+        var provider = transportStub.args[0][0];
+        assert.isOk(provider);
+        assert.equal(provider._authConfig.workloadUri, 'unix:///var/run/iotedge.w.sock');
```

Comment:
>_authConfig [](start = 30, length = 11)

guessing we can't stub the IoTEdgeAuthenticationProvider? i'm not a fan of reaching within private members of another class in unit tests..

Reply:
Not sure how to do that here.